### PR TITLE
407_validated_itl2

### PR DESCRIPTION
**Pull request by okudum**
## Pull Request submission

- Renamed the file to itl.csv  (convert to CSV); placed it in mappers/2023
- 
- Created the schema (a new toml file): all columns, names are in the first row; all dtypes are str
- 
- Read and validate the file in staging/validation module.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes # 407

#### Code

- [x] **Code runs** The code runs on my machine and/or CDSW
- [x] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [x] **Requirements** My/our code functions according to the requirements of the ticket
- [x] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [x] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [x] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/staging/staging_main.py [hunk 1]
```diff
@@ -156,12 +156,18 @@ def run_staging(
...
+    val.validate_data_with_schema(ultfoc_mapper, "./config/ultfoc_schema.toml")
     val.validate_ultfoc_df(ultfoc_mapper)
     StagingMainLogger.info("Foreign Ownership File Loaded Successfully...")
 
+    # Load itl mapper
+    StagingMainLogger.info("Loading ITL File")
+    itl_mapper_path = paths["itl_path"]
+    check_file_exists(itl_mapper_path)
+    itl_mapper = read_csv(itl_mapper_path)
+    val.validate_data_with_schema(itl_mapper, "./config\itl_schema.toml")
```
Comment (dpurches) [inline src/staging/staging_main.py]:
```suggestion
    val.validate_data_with_schema(itl_mapper, "./config/itl_schema.toml")
```

**Review by dpurches — APPROVED** [review]
All looks good and the code runs for me.

Have added a small suggestion just for formatting.

**Comment by dpurches** [comment]
@okudum It wasn't part of the original ticket, but would you be able to rename the `mapper_path` variable here to `pg_mapper` please. Now that we have multiple mappers it doesn't really make sense to call one of them `mapper_path`.

**Comment by okudum** [comment]
> @okudum It wasn't part of the original ticket, but would you be able to rename the `mapper_path` variable here to `pg_mapper` please. Now that we have multiple mappers it doesn't really make sense to call one of them `mapper_path`.

I renamed , Resolved conflict, can you check please?

File: src/staging/staging_main.py [hunk 2]
```diff
@@ -152,9 +152,9 @@ def run_staging(
         StagingMainLogger.info("Loading of Manual Outlier File skipped")
 
     # Load the PG mapper
-    mapper_path = paths["mapper_path"]
-    check_file_exists(mapper_path)
-    mapper = read_csv(mapper_path)
+    pg_mapper = paths["mapper_path"]
```
Comment (dpurches) [inline src/staging/staging_main.py]: Sorry could you also change the name of the argument in the `dev_config` pls. Then the following suggestion should work (just for consistency!)

```suggestion
    pg_mapper = paths["pg_mapper_path"]
```

Comment (okudum) [reply src/staging/staging_main.py]: mapper = read_csv(mapper_path)
  what do you think about "mapper" need to be change as "pg_mapper",  or just leave it as mapper

Comment (dpurches) [reply src/staging/staging_main.py]: Yeh I think change "mapper" to "pg_mapper"

**Comment by dpurches** [comment]
Nice work @okudum! Will merge now

**Pull request merged**